scene/gui/graph_edit.cpp

```diff
@@ -746,9 +746,9 @@ void GraphEdit::_top_layer_input(const Ref<InputEvent> &p_ev) {
 	if (mb.is_valid() && mb->get_button_index() == MouseButton::LEFT && !mb->is_pressed()) {
 		if (connecting_valid) {
 			if (connecting && connecting_target) {
-				String from = connecting_from;
+				StringName &from = connecting_from;
 				int from_slot = connecting_index;
-				String to = connecting_target_to;
+				StringName &to = connecting_target_to;
```

Comment:
Not sure if using reference here is a good idea, because they are swapped below.

Reply:
I'm not completely sure but swapping the `StringName` references should be ok.

Reply:
The issue is these are references to the member fields (`connecting_from`, `connecting_target_to`). Swapping such references will swap the members which can potentially result in bugs as the logic of some other methods referring to these members could get broken. Using `const StringName &` would be fine but it would of course not allow swapping (and that's why it would be fine). So I'd say using plain `StringName` is the way to go.

Or maybe not swapping at all and doing something like this instead:
```cpp
const StringName &from = connecting_out ? connecting_from : connecting_target_to;
int from_port = connecting_out ? connecting_index : connecting_target_index;
const StringName &to = connecting_out ? connecting_target_to : connecting_from;
int to_port = connecting_out ? connecting_target_index : connecting_index;
```
or maybe even something like:
```cpp
if (connecting_out) {
	emit_signal(...);
} else {
	emit_signal(...);
}
```
Anyway, I think it shouldn't be changing the members (like it currently does when swapping).

Reply:
Ok I went with the latter and also simplified the code below. Hopefully I didn't mix the arguments 🙃

Reply:
> Hopefully I didn't mix the arguments

LGTM.